phtree/common/filter.h

```diff
@@ -150,6 +150,132 @@ class FilterAABB {
     const CONVERTER converter_;
 };
 
+/*
+ * The paraboloid filter can be used to query a point tree for an axis-aligned paraboloid.
+ * This is useful for, e.g., doing a range check for a projectile with a fixed launch speed.
+ * Also see https://en.wikipedia.org/wiki/paraboloid_of_safety for this application.
+ */
+template <
+    typename CONVERTER = ConverterIEEE<3>,
+    typename DISTANCE = DistanceEuclidean<3>>
+class FilterParaboloid {
+    using KeyExternal = typename CONVERTER::KeyExternal;
+    using KeyInternal = typename CONVERTER::KeyInternal;
+    using ScalarInternal = typename CONVERTER::ScalarInternal;
+    using ScalarExternal = typename CONVERTER::ScalarExternal;
+
+  public:
+    /*
+     * @param axis The axis along which the paraboloid is aligned.
+     * @param vertex The vertex of the paraboloid.
```

Comment:
Can you descibe what the 'vertex' is? It it the tip, the center of the base, or something else?

Reply:
From what I could find "vertex" seems to be the English word for "Scheitel". I'm happy to change the name if there is a better word.

Reply:
You are right, "vertex" is actually good.